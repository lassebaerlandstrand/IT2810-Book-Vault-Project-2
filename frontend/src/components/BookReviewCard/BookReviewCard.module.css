.card {
  height: 100%;
}

.img {
  display: none;
}

@media screen and (min-width: $mantine-breakpoint-sm) {
  .img {
    display: initial;
  }
}

.textSection {
  text-overflow: ellipsis;
  overflow-wrap: break-word;
}

:where([data-mantine-color-scheme='light']) .card {
  background-color: var(--mantine-color-gray-0);
}

:where([data-mantine-color-scheme='light']) .bookAuthor {
  color: var(--mantine-color-gray-7);
}

:where([data-mantine-color-scheme='dark']) .bookAuthor {
  color: var(--mantine-color-dark-1);
}
