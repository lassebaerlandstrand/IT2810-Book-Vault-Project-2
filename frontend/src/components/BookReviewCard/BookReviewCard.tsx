import { Card, Flex, Grid, Image, Rating, Stack, Text } from '@mantine/core';
import { Review as ReviewType } from '@/generated/graphql';
import styles from './BookReviewCard.module.css';

type ReviewProps = {
  review: ReviewType;
};

/**
 * Card component that displays a single book review with rating and description.
 */
const BookReviewCard = ({ review }: ReviewProps) => {
  if (!review.book) {
    return <Text>Type of ReviewCard is set to bookReview, but user not set</Text>;
  }

  return (
    <Card p={30} radius="lg" className={styles.card} m="auto">
      <Grid align="center">
        <Grid.Col span="content" h={200} className={styles.img}>
          <Image
            src={review.book.coverImg}
            alt={`Cover image for ${review.book.title}`}
            fallbackSrc="https://placehold.co/200x300?text=Cover%20image%20for%20book"
            radius="lg"
            fit="contain"
            w="fit-content"
            maw="100%"
            mah="100%"
            m="auto"
          />
        </Grid.Col>

        <Grid.Col span="auto">
          <Grid justify="flex-start">
            <Grid.Col span="content">
              <Stack gap={0}>
                <Text fw={600} component="h4" lineClamp={2} className={styles.bookTitle}>
                  {`Review of ${review.book.title}`}
                </Text>
                <Text fw={500} size="sm" fs="italic" lineClamp={2}>
                  {new Date(review.at).toDateString()}
                </Text>
              </Stack>
            </Grid.Col>
            <Grid.Col span={12}>
              <Flex justify="left" align="center" gap={7} mt="xs">
                <Rating
                  value={review.rating}
                  fractions={2}
                  readOnly
                  size={review.description && review.description !== '' ? 'md' : 'xl'}
                />
                <Text size={review.description && review.description !== '' ? 'md' : 'xl'}>
                  {review.rating.toFixed(1)}
                </Text>
              </Flex>
            </Grid.Col>
            {review.description && review.description !== '' ? (
              <Grid.Col span={12}>
                <Text lineClamp={5}>{review.description}</Text>
              </Grid.Col>
            ) : (
              <></>
            )}
          </Grid>
        </Grid.Col>
      </Grid>
    </Card>
  );
};

export default BookReviewCard;
