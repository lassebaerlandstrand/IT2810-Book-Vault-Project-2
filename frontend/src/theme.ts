import {
  Container,
  createTheme,
  DEFAULT_THEME,
  Loader,
  MantineColorsTuple,
  rem,
} from '@mantine/core';
import { BookLoader } from './components/Loading/Loading';

const CONTAINER_SIZES: Record<string, string> = {
  xxs: '24em',
  xs: '30em',
  sm: '48em',
  md: '64em',
  lg: '75em',
  xl: '90em',
};

// https://colorkit.co/color-palette-generator/02070b-030a11-030d16-051321-06192b-092641-0c3256-0f3f6c-124b81-155896/
const darkBlue: MantineColorsTuple = [
  '#155896',
  '#124b81',
  '#0f3f6c',
  '#0c3256',
  '#092641',
  '#06192b',
  '#051321',
  '#030d16',
  '#030a11',
  '#02070b',
];

// https://colorkit.co/color-palette-generator/6a5f4a-d4bd94-dac5a1-e2d1b5-e9ddc8-eee4d3-f2eade-f6f0e7-f9f5ef-fcfaf7/
const beige: MantineColorsTuple = [
  '#fcfaf7',
  '#f9f5ef',
  '#f6f0e7',
  '#f2eade',
  '#eee4d3',
  '#e9ddc8',
  '#e2d1b5',
  '#dac5a1',
  '#d4bd94',
  '#6a5f4a',
];

export const theme = createTheme({
  fontFamily: `Poppins, ${DEFAULT_THEME.fontFamily}`,
  breakpoints: {
    xxs: '24em', // 384px
    xs: '30em', // 480px
    sm: '48em', // 768px
    md: '64em', // 1024px
    lg: '75em', // 1200px
    xl: '90em', // 1440px
  },
  colors: {
    'dark-blue': darkBlue,
    beige,
  },
  components: {
    Container: Container.extend({
      vars: (_, { size, fluid }) => ({
        root: {
          '--container-size': fluid
            ? '100%'
            : size !== undefined && size in CONTAINER_SIZES
              ? CONTAINER_SIZES[size]
              : rem(size),
        },
      }),
    }),
    Loader: Loader.extend({
      defaultProps: {
        loaders: { ...Loader.defaultLoaders, book: BookLoader },
      },
      vars: (_, props) => {
        if (props.size === 'xxl') {
          return {
            root: {
              '--loader-size': rem(90),
            },
          };
        }
        return { root: {} };
      },
    }),
  },
});
