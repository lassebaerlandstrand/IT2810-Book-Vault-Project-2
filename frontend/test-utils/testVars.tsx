import { Book } from '@/generated/graphql';

export const dummyBook: Book = {
  __typename: 'Book',
  authors: [{ __typename: 'Author', name: 'A name' }],
  coverImg: 'https://placehold.co/200x300?text=Cover%20image%20for%20book',
  genres: [{ __typename: 'Genre', name: 'A genre' }],
  id: '1',
  publisher: { __typename: 'Publisher', name: 'A publisher' },
  title: 'A title',
  bookFormat: 'Paper',
  characters: ['A character'],
  description: 'A deep book about deep topics',
  isbn: '999999999',
  language: 'Swedish',
  pages: 1000,
  publishDate: '03/03/03',
  rating: 3.555,
  series: 'Title #1',
  numRatings: 5000,
  ratingsByStars: [1, 5, 10, 7, 1],
};
