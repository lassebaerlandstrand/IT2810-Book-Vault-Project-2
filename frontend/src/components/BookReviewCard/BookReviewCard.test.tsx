import { render, screen } from '@testing-library/react';
import { MantineProvider } from '@mantine/core';
import { theme } from '@/theme';
import { dummyBook } from '../../../test-utils/testVars';
import BookReviewCard from './BookReviewCard';

const dummyReview = {
  UUID: 'review-1',
  rating: 2,
  description: 'Great book!',
  at: '2023-10-30T12:00:00Z',
};

describe('BookReviewCard Component', () => {
  test('shows error when missing book', async () => {
    render(
      <MantineProvider theme={theme}>
        <BookReviewCard review={dummyReview} />
      </MantineProvider>
    );
    expect(
      screen.getByText('Type of ReviewCard is set to bookReview, but user not set')
    ).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MantineProvider theme={theme}>
        <BookReviewCard review={{ ...dummyReview, book: dummyBook }} />
      </MantineProvider>
    );

    const attributesToRemove = [
      ...document.body.querySelectorAll('div [id^="mantine"]'),
      ...document.body.querySelectorAll('div [for^="mantine"]'),
    ]; // Because Mantine uses random ids which causes snapshots to fail
    attributesToRemove.forEach((element) => {
      element.removeAttribute('for');
      element.removeAttribute('id');
      element.removeAttribute('class');
      element.removeAttribute('aria-describedby');
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
