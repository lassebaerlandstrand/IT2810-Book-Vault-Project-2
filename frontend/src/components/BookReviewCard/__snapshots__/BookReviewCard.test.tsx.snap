// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BookReviewCard Component > matches snapshot 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-font-family: Poppins, Roboto, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji;--mantine-font-family-headings: Poppins, Roboto, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji;--mantine-breakpoint-xs: 30em;--mantine-breakpoint-md: 64em;--mantine-breakpoint-xl: 90em;--mantine-breakpoint-xxs: 24em;--mantine-color-logo-theme-0: #fefefe;--mantine-color-logo-theme-1: #cbcfd2;--mantine-color-logo-theme-2: #97a0a6;--mantine-color-logo-theme-3: #7d8991;--mantine-color-logo-theme-4: #63717b;--mantine-color-logo-theme-5: #495a65;--mantine-color-logo-theme-6: #2f434f;--mantine-color-logo-theme-7: #132937;--mantine-color-logo-theme-8: #0b202e;--mantine-color-logo-theme-9: #061017;}:root[data-mantine-color-scheme="dark"]{--mantine-color-logo-theme-text: var(--mantine-color-logo-theme-4);--mantine-color-logo-theme-filled: var(--mantine-color-logo-theme-8);--mantine-color-logo-theme-filled-hover: var(--mantine-color-logo-theme-9);--mantine-color-logo-theme-light: rgba(47, 67, 79, 0.15);--mantine-color-logo-theme-light-hover: rgba(47, 67, 79, 0.2);--mantine-color-logo-theme-light-color: var(--mantine-color-logo-theme-3);--mantine-color-logo-theme-outline: var(--mantine-color-logo-theme-4);--mantine-color-logo-theme-outline-hover: rgba(99, 113, 123, 0.05);}:root[data-mantine-color-scheme="light"]{--mantine-color-logo-theme-text: var(--mantine-color-logo-theme-filled);--mantine-color-logo-theme-filled: var(--mantine-color-logo-theme-6);--mantine-color-logo-theme-filled-hover: var(--mantine-color-logo-theme-7);--mantine-color-logo-theme-light: rgba(47, 67, 79, 0.1);--mantine-color-logo-theme-light-hover: rgba(47, 67, 79, 0.12);--mantine-color-logo-theme-light-color: var(--mantine-color-logo-theme-6);--mantine-color-logo-theme-outline: var(--mantine-color-logo-theme-6);--mantine-color-logo-theme-outline-hover: rgba(47, 67, 79, 0.05);}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 29.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 30em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 63.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 64em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 89.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 90em) {.mantine-hidden-from-xl {display: none !important;}}@media (max-width: 23.99375em) {.mantine-visible-from-xxs {display: none !important;}}@media (min-width: 24em) {.mantine-hidden-from-xxs {display: none !important;}}
  </style>
  <div
    class="_card_08c62f m_e615b15f mantine-Card-root m_1b7284a3 mantine-Paper-root"
    style="--paper-radius: var(--mantine-radius-lg); margin: auto;"
  >
    <style
      data-mantine-styles="inline"
    >
      .__m__-r2{--grid-gutter:var(--mantine-spacing-md);}
    </style>
    <div
      class="m_410352e9 mantine-Grid-root __m__-r2"
    >
      <div
        class="m_dee7bd2f mantine-Grid-inner"
      >
        <style
          data-mantine-styles="inline"
        >
          .__m__-r4{--col-flex-grow:auto;--col-flex-basis:auto;--col-width:auto;--col-max-width:unset;}
        </style>
        <div
          class="m_96bdd299 mantine-Grid-col _img_08c62f __m__-r4"
        >
          <img
            alt="Cover image for A title"
            class="m_9e117634 mantine-Image-root"
            src="https://placehold.co/200x300?text=Cover%20image%20for%20book"
            style="--image-radius: var(--mantine-radius-lg); --image-object-fit: contain; margin: auto; max-width: 100%; max-height: 100%;"
          />
        </div>
        <style
          data-mantine-styles="inline"
        >
          .__m__-r7{--col-flex-grow:1;--col-flex-basis:0rem;--col-max-width:100%;}
        </style>
        <div
          class="m_96bdd299 mantine-Grid-col __m__-r7"
        >
          <style
            data-mantine-styles="inline"
          >
            .__m__-r9{--grid-gutter:var(--mantine-spacing-md);}
          </style>
          <div
            class="m_410352e9 mantine-Grid-root __m__-r9"
            style="--grid-justify: flex-start;"
          >
            <div
              class="m_dee7bd2f mantine-Grid-inner"
            >
              <style
                data-mantine-styles="inline"
              >
                .__m__-rb{--col-flex-grow:auto;--col-flex-basis:auto;--col-width:auto;--col-max-width:unset;}
              </style>
              <div
                class="m_96bdd299 mantine-Grid-col __m__-rb"
              >
                <div
                  class="m_6d731127 mantine-Stack-root"
                  style="--stack-gap: 0rem; --stack-align: stretch; --stack-justify: flex-start;"
                >
                  <h4
                    class="mantine-focus-auto _bookTitle_08c62f m_b6d8b162 mantine-Text-root"
                    data-line-clamp="true"
                    style="--text-line-clamp: 2; font-weight: 600;"
                  >
                    Review of A title
                  </h4>
                  <p
                    class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                    data-line-clamp="true"
                    data-size="sm"
                    style="--text-fz: var(--mantine-font-size-sm); --text-lh: var(--mantine-line-height-sm); --text-line-clamp: 2; font-weight: 500; font-style: italic;"
                  >
                    Mon Oct 30 2023
                  </p>
                </div>
              </div>
              <style
                data-mantine-styles="inline"
              >
                .__m__-rg{--col-flex-grow:auto;--col-flex-basis:auto;--col-width:auto;--col-max-width:unset;}
              </style>
              <div
                class="m_96bdd299 mantine-Grid-col __m__-rg"
              >
                <div
                  class="m_8bffd616 mantine-Flex-root __m__-ri"
                  style="gap: calc(0.4375rem * var(--mantine-scale)); align-items: center; justify-content: center;"
                >
                  <div
                    data-size="sm"
                    style="--rating-size: var(--rating-size-sm); --rating-color: var(--mantine-color-yellow-filled);"
                  >
                    <div
                      class="m_61734bb7 mantine-Rating-symbolGroup"
                    >
                      <div
                        data-read-only="true"
                        style="--rating-item-z-index: 0;"
                      >
                        <div
                          class="m_fae05d6a mantine-Rating-symbolBody"
                          style="--rating-symbol-clip-path: inset(0 100% 0 0);"
                        >
                          <svg
                            class="m_5662a89a mantine-Rating-starSymbol"
                            data-filled="true"
                            fill="none"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
                            />
                          </svg>
                        </div>
                      </div>
                      <div
                        data-read-only="true"
                        style="--rating-item-z-index: 0;"
                      >
                        <div
                          class="m_fae05d6a mantine-Rating-symbolBody"
                          style="--rating-symbol-clip-path: inset(0 100% 0 0);"
                        >
                          <svg
                            class="m_5662a89a mantine-Rating-starSymbol"
                            data-filled="true"
                            fill="none"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
                            />
                          </svg>
                        </div>
                      </div>
                      <div
                        data-read-only="true"
                      >
                        <div
                          class="m_fae05d6a mantine-Rating-symbolBody"
                        >
                          <svg
                            class="m_5662a89a mantine-Rating-starSymbol"
                            data-filled="true"
                            fill="none"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <div
                      class="m_61734bb7 mantine-Rating-symbolGroup"
                    >
                      <div
                        data-read-only="true"
                        style="--rating-item-z-index: 0;"
                      >
                        <div
                          class="m_fae05d6a mantine-Rating-symbolBody"
                          style="--rating-symbol-clip-path: inset(0 100% 0 0);"
                        >
                          <svg
                            class="m_5662a89a mantine-Rating-starSymbol"
                            data-filled="true"
                            fill="none"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
                            />
                          </svg>
                        </div>
                      </div>
                      <div
                        data-read-only="true"
                      >
                        <div
                          class="m_fae05d6a mantine-Rating-symbolBody"
                        >
                          <svg
                            class="m_5662a89a mantine-Rating-starSymbol"
                            data-filled="true"
                            fill="none"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <div
                      class="m_61734bb7 mantine-Rating-symbolGroup"
                    >
                      <div
                        data-read-only="true"
                        style="--rating-item-z-index: 0;"
                      >
                        <div
                          class="m_fae05d6a mantine-Rating-symbolBody"
                          style="--rating-symbol-clip-path: inset(0 100% 0 0);"
                        >
                          <svg
                            class="m_5662a89a mantine-Rating-starSymbol"
                            fill="none"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
                            />
                          </svg>
                        </div>
                      </div>
                      <div
                        data-read-only="true"
                      >
                        <div
                          class="m_fae05d6a mantine-Rating-symbolBody"
                        >
                          <svg
                            class="m_5662a89a mantine-Rating-starSymbol"
                            fill="none"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <div
                      class="m_61734bb7 mantine-Rating-symbolGroup"
                    >
                      <div
                        data-read-only="true"
                        style="--rating-item-z-index: 0;"
                      >
                        <div
                          class="m_fae05d6a mantine-Rating-symbolBody"
                          style="--rating-symbol-clip-path: inset(0 100% 0 0);"
                        >
                          <svg
                            class="m_5662a89a mantine-Rating-starSymbol"
                            fill="none"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
                            />
                          </svg>
                        </div>
                      </div>
                      <div
                        data-read-only="true"
                      >
                        <div
                          class="m_fae05d6a mantine-Rating-symbolBody"
                        >
                          <svg
                            class="m_5662a89a mantine-Rating-starSymbol"
                            fill="none"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <div
                      class="m_61734bb7 mantine-Rating-symbolGroup"
                    >
                      <div
                        data-read-only="true"
                        style="--rating-item-z-index: 0;"
                      >
                        <div
                          class="m_fae05d6a mantine-Rating-symbolBody"
                          style="--rating-symbol-clip-path: inset(0 100% 0 0);"
                        >
                          <svg
                            class="m_5662a89a mantine-Rating-starSymbol"
                            fill="none"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
                            />
                          </svg>
                        </div>
                      </div>
                      <div
                        data-read-only="true"
                      >
                        <div
                          class="m_fae05d6a mantine-Rating-symbolBody"
                        >
                          <svg
                            class="m_5662a89a mantine-Rating-starSymbol"
                            fill="none"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                  <p
                    class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                    style="font-weight: 500;"
                  >
                    2.0
                  </p>
                </div>
              </div>
              <style
                data-mantine-styles="inline"
              >
                .__m__-r1e{--col-flex-grow:auto;--col-flex-basis:100%;--col-max-width:100%;}
              </style>
              <div
                class="m_96bdd299 mantine-Grid-col __m__-r1e"
              >
                <p
                  class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                  data-line-clamp="true"
                  style="--text-line-clamp: 5;"
                >
                  Great book!
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
